### ISSUE TITLE
#set directive: Assignment value lost after foreach loop

### ISSUE THREAD

**Issue by flofuchs** [purcell/airspeed#17]
Variables that are created via #set *prior* to a foreach block are readable from within that block, but any assignments to that variable are forgotten *after* the foreach block. Here's a test case (https://github.com/flofuchs/airspeed/commit/bc1c2807a57bd39b73ea188d5d846617521763fe) that illustrates the behaviour:

```python
def test_outer_variable_assignable_from_foreach_block(self):
    template = airspeed.Template(
        "#set($var = 1)"
        "#foreach ($i in $items)$var,#set($var = $i)#end"
        "$var")
    self.assertEquals("1,2,3,4", template.merge({"items": [2, 3, 4]}))
```

```python
AssertionError: '1,2,3,4' != u'1,2,3,1'
```

Using the original Java engine `$var` is rendered as `4` after the loop.

Other than that: Thanks for doing a Python implementation of Velocity. It's been helping our team greatly in a long-term project. 

Florian

**Comment by purcell** [comment]
What should the behaviour of this be? -

```
#set($i = 1)
#foreach ($i in [1,2,3])#end
$i
```

**Comment by purcell** [comment]
If I know that, I can figure out how best to fix it. I don't have an official Velocity lying around to test it with. :-)

**Comment by flofuchs** [comment]
Hey, thanks for the quick answer.

I just ran a quickt test with the official engine: In this case the variable outside the foreach block stays untouched is rendered with the same value as above the foreach block. Like in this test case (which currently succeeds): https://github.com/flofuchs/airspeed/commit/a4b763ce42554c9438d25339b6f633b6df7d39c2

**Comment by purcell** [comment]
Done, and released in 0.5.2. Let me know if you experience any issues.

**Comment by flofuchs** [comment]
Awesome, thanks a ton!
